File: frontend/src/types/agent.ts
// 智能体模块类型定义

// 智能体
export interface Agent {
  id: string
  name: string                    // 智能体名称
  desc: string                    // 描述
  model: string                   // 模型
  prompt: string                  // 系统提示词
  temp: number                    // 温度（0-2）
  maxtok: string                  // 最大 Token 数
  tools: string[]                 // 挂载的工具 ID 列表
  docs: string[]                  // 挂载的文档 ID 列表
  wfs: string[]                   // 挂载的工作流 ID 列表
  mcps: string[]                  // 挂载的 MCP 服务 ID 列表
  skills: string[]                // 挂载的技能 ID 列表
  enabled: boolean                // 是否启用
  lastActive: string              // 最近活跃时间
  createdAt?: string             // 创建时间
}

File: backend/app/schemas/agent.py
"""智能体模块 Pydantic schema（对齐前端 types/agent.ts）。"""
from pydantic import BaseModel


class AgentCreate(BaseModel):
    name: str
    desc: str = ""
    model: str = "gpt-4o"
    prompt: str = ""
    temp: float = 0.7
    maxtok: str = "2048"
    tools: list[str] = []
    docs: list[str] = []
    wfs: list[str] = []
    mcps: list[str] = []
    skills: list[str] = []
    enabled: bool = True


class AgentUpdate(BaseModel):
    name: str | None = None
    desc: str | None = None
    model: str | None = None
    prompt: str | None = None
    temp: float | None = None
    maxtok: str | None = None
    tools: list[str] | None = None
    docs: list[str] | None = None
    wfs: list[str] | None = None
    mcps: list[str] | None = None
    skills: list[str] | None = None
    enabled: bool | None = None


class AgentOut(BaseModel):
    id: str
    name: str
    desc: str
    model: str
    prompt: str
    temp: float
    maxtok: str
    tools: list[str]
    docs: list[str]
    wfs: list[str]
    mcps: list[str]
    skills: list[str]
    enabled: bool
    lastActive: str
    createdAt: str | None = None
